sdk/metric/pipeline.go

```diff
@@ -133,7 +133,8 @@ func (p *pipeline) produce(ctx context.Context) (metricdata.ResourceMetrics, err
 			errs.append(err)
 		}
 		if err := ctx.Err(); err != nil {
-			return metricdata.ResourceMetrics{}, err
+			*rm = metricdata.ResourceMetrics{}
```

Comment:
Is this needed?

The next time the passed `rm` is used it will need to re-allocate. Should it just be left as is?

Reply:
There should be some cleanup so that we don't leave a half-allocated `rm` sitting around.  Would you rather see a resizing ScopeMetrics to zero?

Reply:
> Would you rather see a resizing ScopeMetrics to zero?

Yeah, that sounds good. :+1:

(that and setting the resource to nil sounds good to me)

That way the caller could still reuse `rm` without having to allocate again.